### ISSUE TITLE
Magnet links not launching, torrent not found..

### ISSUE THREAD

**Issue by cocokola** [SchizoDuckie/DuckieTV#892]
**What build of DuckieTV are you using (Standlone / Chrome Extension (New Tab / Browser Action))**
...Standalone, nightly 17th and todays.

**What version of DuckieTV are you using (Stable 1.1.x / Nightly yyyymmddHHMM)**
...	201704210130

**What is your Operating System (Windows, Mac, Linux, Android)**
...windows

**Describe the problem you are having and steps to reproduce if available**
...Gold Rush : Parker's trail

**Attach any DuckieTV statistics or Developer Console logs if available**
...
![image](https://cloud.githubusercontent.com/assets/9909697/25259730/7943abb2-260d-11e7-8408-745dcb53fc9b.png)

[bagblfnohpiddokbpaacecffdboomafe-1492740562060.txt](https://github.com/SchizoDuckie/DuckieTV/files/945885/bagblfnohpiddokbpaacecffdboomafe-1492740562060.txt)

Strange things, I'll learn from this one.  I seem to run into many torrents that are not found, removing the season/episode after clicking on search for the episode, brings up what I want. 
Problem #1, I have this with SO many torrents.  My worst problem is with Fast N loud - it shows s10e10 on the calendar but click on search and it shows s10e16?
![f lcapture](https://cloud.githubusercontent.com/assets/9909697/25260009/16c8bcb4-260f-11e7-931c-923c1ef7be50.JPG)
Others like this are street outlaws, The Voice, American Pickers.. 

Problem#2   If I use "magnet link" I can then go to Deluge and click add and it works.  If I click download magnet (the way it should work) then nothing (see log error showing no such hash.)  If I open WebUI on my browser, THEN everything works.  I changed the port and password, but explain why does it show I am connected?  I just can't launch the torrent.   I can't leave WebUI running, as it crashes after about 30 minutes to an apparent memory leak in Chrome, latest offical build.  I see event windows 'starting ... ' frequently, but it will not work.  It shows it is connected to Deluge when I check the connection, I have a green dot on the left menu.

a HUGE thank you for the amazing product and support.

**Comment by garfield69** [comment]
>Problem #1, I have this with SO many torrents. My worst problem is with Fast N loud - it shows s10e10 on the calendar but click on search and it shows s10e16?    
>Others like this are street outlaws, The Voice, American Pickers..     

Actually, if you look at the screen capture image you provided, it's s07e16.     
Believe it or not, this is actually working as intended :-)     
As you may know, there are shows whose season and episode numbers as broadcast by a network and/or recorded on crowd sourced TV sites like *thetvdb.com*, that end up with torrents on the *torrent scene* with different naming schemes.     
Sometimes this is simply due because the episode due to be aired is skipped or is aired back to back with the next episode or whatever. Other times it's more complicated. But the result is a disconnect in the sequence numbering between what the TV schedule lists and what the torrent names on the *scene* are labelled.      
To keep track of these differences, DuckieTV uses another crowd sourced site called *theXEM.de*.     
They maintain a cross entity map for shows that they monitor, and DuckieTV fetches a table update daily to keep as up to date as it can.     
All the shows you mentioned have entries on theXem.de and the season/episode returned are what they believe the torrent scene is currently using for that particular episode.      

>Problem#2 If I use "magnet link" I can then go to Deluge and click add and it works. If I click download magnet (the way it should work) then nothing (see log error showing no such hash.) If I open WebUI on my browser, THEN everything works. I changed the port and password, but explain why does it show I am connected?     

This is an actual problem.     
The log you provided shows that while Deluge is *Connected*, it is not authorised.      
As you may know, The DuckieTV Deluge client posts a simple connect request to your Deluge Host and if it gets a response, DuckieTV shows a green dot and reports it is connected.      
However, this is akin to someone having dialled a mobile number and someone at the other end pressing the green answer button. We have a connection, but this is not quite the same as reporting that both ends are *communicating*.     
DuckieTV's Deluge client then asks for a list of torrents that the Deluge Host has, and the log shows that this is failing. Usually this is because the Deluge Host has determined that the DuckieTV Deluge Client is not authorised to request such a list.     

To progress this further, we are going to need to know which version of Deluge you are using. There was a recent problem https://github.com/SchizoDuckie/DuckieTV/issues/880 where Deluge made some changes that broke our Deluge Client protocol.     
The current Stable release of Deluge is 1.3.14 and the latest nightly is set to talk with this version correctly. Upgrade your Deluge to this version if you are not on it.     
If you are already on this version, then let us know and we move on to the next stage in our diagnosis.

**Comment by cocokola** [comment]
I have so much respect for you for explaining everything;thank you for
taking the time. Problem number 1,  for the shows I listed, I'm not getting
any hits using all the  torrent search engines.. So what do you suggest I
do? I'll review your comments to understand.  I was impressed with your
exposing the console f12.. I'm not a programmer, but I can follow coding
although I'm old school... I'm into scripting /automation for Windows;
 former Windows active directory architect/engineer.

I'm using the latest deluge client, I saw your note about the conflict.
I've had this problem since I first set things up about 8 weeks ago.  I
assumed webui browser needed up be running at first..

So I've had production release of duckytv and the older deluge, to
troubleshoot before opening a ticket, I upgraded deluge to latest, and
duckytv to latest stand alone nightly.  Same problem, opened the ticket
with you.  Let me know how I can help, I'll gladly learn if you have links
to share, and I'm pleased to help flush out the bug.

Based on your description, you are opening a connection to the webui,
getting a response;however, since your test connection can verify password,
you are doing a handshake.

I'll offer a possible source : I suspect the problem may be unique to folks
using vpn who leverage (in my case Windows advanced)  firewall to ensure if
the vpn drops then deluge is prevented from accepting inbound /outbound
connections.  This is accomplished by setting a deny all on the ports used
for deluge,  then allow the deluge.exe process to open ports for
destination ipaddress in the 10.x range only.  Local lan is a class c 192.x

I'll admit, I've had some difficulty setting this up properly initially (I
read fifty articles that suggested using public/private zones on windows to
no avail.) it is working, but perhaps this could be an area of focus?

I was going to change the rules anyway to allow my mobile device to use
transender to manage deluge (no matter if vpn is up or down), so it's good
timing.  What doesn't make sense is how can you get an auth handshake but
not interact?  Same ports and same method are used, correct?

Thanks!

**Comment by cocokola** [comment]
As far as problem#1, you are correct, I neglected to show that was replaced as expected - however, this filename is showing 0 results, as are the other shows I  mentioned, for ALL episodes for all those shows, no torrents are found.  The only way I can find them is by erasing the E/S lookup with a manual search..  but it does the lookup again, which then terminates.  That is NOT how I would expect it to work, do you?  The manual search brings up the KNOWN name that is available.  Why would you do another lookup?  Can you add a toggle to override this lookup when manually searching?   Or, add logic so if 0 results are found (no hash found) then go with the non-look up code?  I would suspect this same logic would work for the automatic search: if "none found", give me the option to skip the translation step of thexem?  I don't care if it is an option I can override, OR you automatically select it as a backup.  The worse case is you download the wrong show, vs never downloading any show, which is what I have for the majority of shows I am looking..

I will check into TheXEM, see why they think the known name is obviously wrong.  Perhaps this is an attack from "those" who wish to break the system from working?  I'll see what they do, obviously, they are wong for ALL the search engines you have coded.  Seems suspect, no?

again, thanks!

**Comment by garfield69** [comment]
Re: Deluge.     
OK, vpn and firewall complicate matters, but lets work through what we know so far.     
I am presuming that your Deluge Host and DuckieTV are both running on the same Windows computer, and that you are using either localhost or 127.0.0.1 as the address to access the Deluge Host WebUI from both the DuckieTV Deluge Client and your Chrome Browser. Is this correct?     

You previously stated that if you login to Deluge by using the Browser Deluge WebUI, then you are able to subsequently get DuckieTV to submit magnets to your Deluge Host successfully, using the magnet button on the DuckieTV torrent search results dialogue.      
So my working theory at the moment is that your current DuckieTV Deluge Client setup has an incorrect setting.     

To eliminate this I would like to try the following:      
- logoff the Browser WebUI session to your Deluge Host, and close that browser tab.      
- shutdown DuckieTV and Deluge.      
- Double check that both Deluge and DuckieTV are really gone, by using the Task Manager to check not just that the apps are not active, but that there are no processes running either (The DuckieTV ones may be listed as NWJS processes).      
- startup Deluge      
- startup DuckieTV      
- go to the DuckieTV *Settings, Deluge Integration* panel and click on the *Test and Save* button (This performs a login to verify that the address, port and password are correct).         
- confirm that the status on the Deluge Integration panel reads `Status: Connected to Deluge Web API @ http://localhost:8112` (or whatever the address and port are that you are using).      
- if the status is `Status: Not connected Error: Deluge Connect call failed.` then clear the password field and type in the current password you have been using when you login via the Browser WebUI, and check that the address and port are the same as well. Click the Test and Save button again.      

Notes: If you change the listening port on the Deluge Host preferences WebUI panel, then make sure to recycle both the Deluge Host and DuckieTV.      
The Deluge Host does not close the old listening port when it opens the new one, and his can cause confusion.


Re: thexem.de and torrent searches. I need to think on this a bit and see if I can figure out what might be going on, especially now I know that vpn and firewalls are involved.

**Comment by cocokola** [comment]
Bug? problem #2
a) Clarification:  "You previously stated that if you login to Deluge by
using the Browser Deluge WebUI, then you are able to subsequently get
DuckieTV to submit magnets to your Deluge Host successfully, using the
magnet button on the DuckieTV torrent search results dialogue."
not exactly.  Actually, what happens is I launch deluge, then duckytv.  I
see Duckytv tries to launch magnets, but nothing happens.

If I start up a webui in a browser, login, click connect, then within
seconds, shows start populating in Deluge.  The information on manually
launching was just to let you know I can do a magnet link (requiring me to
click on add on deluge)  but not a magnet download.

b) Your suggestion for a clean start is exactly what I did during testing.
I followed your suggestion anyway, and it shows: Status: Connected to Deluge
Web API @ http://localhost:7514
yes I am using 127.0.0.1

So basically, I am right where I left off previously.
Note: deluge works as expected if I bypass duckytv and just download a
magnet link. Further thinking, I don't see how VPN/firewall rules could
have anything to do with the problem where launching webui would fix.  I am
reviewing the firewall rules

c) Firewall rules:
for process named Deluge(d, console, gui,new)
        outbound: block <gateway IP>
        inbound: allow (tcp/udp webui port)
                     Process deluge(d console, gui)
                               , allow  Dest IP <VPN range>
                                 block dest IP <gateway IP>


Problem #1
a) I don't see how VPN/firewall has anything to do with "thexem.de"
translation.
Do you not see a benefit to basically give the user an option to allow
ignoring the translation? or I prefer, the user have an option if a torrent
is not found, to try pulling torrent that has not been translated?

b) I noticed I have shows that have been downloaded and completed; however,
in the calendar, the show is not green/checked.   I see the SE is different
than actual (I assume translated.) Is this by design?  It's hard to tell
what is not working vs. is working via the calendar..

thanks

**Comment by garfield69** [comment]
re: Deluge      
well its a puzzle. If you use the Test&Save button on the DuckieTV Deluge integration panel, and you get a successful connection, then there is no reason why you should not be able to have DuckieTV submit magnets to your Deluge Host via the DuckieTV Deluge WebUI interface.      
This works on my win7 sandbox perfectly well, using Deluge 1.3.14 and Nightly 20170422.     
Also, We have not received any other Deluge users reporting problems so far.     

Magnet **links** are different, in that they bypass the interface. On the Standalone DuckieTV, we get NWJS to issue a `nw.Shell.openExternal(url)` which invokes the Windows OS to do the handover.     

The only conclusion I can draw at this stage, is that something in your Windows Environment is interfering with the DuckieTV Deluge Client WebUI connection to the Deluge Host.     
Since the problem does not appear to be with the DuckieTV **code** that I can see, I am not sure that I can be much further help in resolving this with you.      
Have you tried using an alternate torrent host? Perhaps a test using uTorrent, or one of the others may yield a clue to what is going on for you?      

re:Torrent Search with no results.     
Looking back at your log, I see many http status 403 (Forbidden). This is more likely the reason for the no-results you are seeing on your torrent searches.     
The whole point of using XEM is to provide the best match between the torrent scene and the broadcast s/e. Without it, you are almost always going to either: 1. not find a torrent with the broadcast s/e name because the scene is not naming torrents that way, or 2. you will find a torrent with that name but it will not contain the episode you are after.    
I do not see any benefit to implementing an option setting that turns off XEM use.      
But I will of course defer to @SchizoDuckie  on this, as the author he has the final word.

**Comment by cocokola** [comment]
Well, this is what I am greeted with:   nothing found for 80% of the
torrents..

I believe you aren't keeping your mind open.  Your reply is as if i
suggested bypass XEM.  I am not.  Please re-read my response.  I am asking
for an option to be added.  I am asking to add an option with a days
setting.  If enabled, for "Not found" torrents how many days to wait after
an episode aired (value from calendar- default say 5 days?.).Then ADD
additional code that checks: if  "not found response" for torrent (no hash
found), if the current date is (xx) days after, do a search for the show
WITHOUT the XEM look-up.

First implementation, skip the xx days feature.

 Perhaps there is a reason you need the OEM lookup?  I never asked you to
SKIP the lookup on the first try, just upon ONE exception. Make more sense?
....................

also, Can you help me verify if there is a problem with the data on XEM or
something else?


Finally, I eliminated the VPN and the firewall rules.  I am getting all the
same results.

So, ask yourself, what does having webui open and running

Where can I find the autodownload lookup logging?  maybe I can narrow it
down myself, but I'll definitely take pointers!

thanks,

[image: Inline image 1]

**Comment by garfield69** [comment]
Ok, so let put aside the xem topic for now.     

>also, Can you help me verify if there is a problem with the data on XEM or something else?      

Sure. Lets try this. zip up your DuckieTV data directory and place it in the cloud (say Google Drive, for instance), post a share link here and I will load my sandbox with your data and settings.     
Then we can compare results from searches.     
You should find your data directory at this location: `C:\Users\<username>\AppData\Local\DuckieTV-Standalone`     
Pick two or three different series, and focus on a recent episode for each, and do a search and post a screencapture of your results. I will do the same afterwards and we can compare.     

>Where can I find the autodownload lookup logging?  maybe I can narrow it down myself, but I'll definitely take pointers!     

Perhaps you might find the following wiki useful: https://github.com/SchizoDuckie/DuckieTV/wiki/Auto-Download-FAQs     
Then if you have any questions I can try to fill in the blanks.

**Comment by cocokola** [comment]
​​
On Sun, Apr 23, 2017 at 12:57 AM, garfield69 <notifications@github.com>
wrote:

> C:\Users\<username>\AppData\Local\DuckieTV-Standalone

For shiggles, I tried adding some random torrents, placed deluge in pause
modeso no downloads would actually pull. Here is some findings.

With VPN up or down, the client is able to connect, and CAN see what is
downloaded, so this is a specific issue to lauching magnets only.

0) The Expanse S00E21  No results found for
*This is actually "inside the expanse: season 2 episode 11  *

*I don't have an account with xem and registration is closed.  I'm on IRC
to see if someone else can create the content.  In IRC they said a The
Expanse was not created as there isn't a problem online with naming.
Inside the Expanse is another issue, I am waiting to see what the the
admins say how to handle.  *
*A user on IRC said to switch to sonarr - as it handles these shows
properly. I may take a look just to see what it does if nothing else.*

*1) The Voice *
*Manual search show files as "the.voice.us.s12e14..... or
The.Voice>US.S11.xxxxx *
*XEM table is three digits for each E and S, although a 1 digit difference
between DBs.  *
*System shows  "Nothing found'  and the search is shown as "The Voice US
S12E13"  *

*2)American Pickers*
*none found, but it is showingf as seasion 0, what is available is season
16. XEM data exists, but the lookup isn't working. *

*3a) With VPN on (default for most users, no?) I checked **the console,
there was "failed to load resource" for zoogle.  I manually copied and
pasted the link and I was welcomed with a robot check 'click to prove I was
human.'     I am getting through once I prove I am human. Doing a check
manually does not fix the app's ability to connect.  Seems like a design
flaw. *

*3b)Skytorrents provided no response at all, until VPN was down. I tried 4
VPN locations and no dice.  VPN down it responded fine.   .*

*with VPN down, I caught the following event:*
Firing magnet URI!
 magnet:?xt=urn:btih:4000ad1fd8d45aa0f9db0a11df4f422f35bac80d&dn=Deadliest.Catch.S00E35.Legends.Born.and.Broken.1080p.DISC.WEBRip.AAC2.0.x264-NTb%5Brartv%5D&tr=http%3A%2F%
2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me
%3A2710&tr=udp%3A%2F%2F9.rarbg.to%3A2710&tr=udp%3A%2F%2Fopen.demonii.com%3A1337%2Fannounce
6045385 null null
deps.js:2433 TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at app.js:12343
    at deps.js:2446
    at m.$eval (deps.js:2461)
    at m.$digest (deps.js:2458)
    at m.$apply (deps.js:2461)
    at l (deps.js:2413)
    at completeRequest (app.js:401)
    at XMLHttpRequest.requestLoaded (app.js:342)
(anonymous) @ deps.js:2433
(anonymous) @ deps.js:2406
(anonymous) @ deps.js:2447
$eval @ deps.js:2461
$digest @ deps.js:2458
$apply @ deps.js:2461
l @ deps.js:2413
completeRequest @ app.js:401
requestLoaded @ app.js:342
deps.js:2433 Error executing Deluge getTorrents

*could this be part of the problem?*

*4)VPN up console output while I started the app and ran manual searches:*
*Xem does not have episode S00E11 for American Pickers (141181), using
default format.app.js:10811 Xem does not have episode S2017E06 for American
Pickers (141181), using default format.app.js:10811 Xem does not have
episode S00E12 for American Pickers (141181), using default
format.app.js:10811 Xem does not have episode S00E10 for American Pickers
(141181), using default format.app.js:10811 Xem does not have episode
S12E18 for The Voice (247824), using default format.app.js:10806 Xem has
episode S10E10 for Fast N' Loud (259393), using mapped format.
Objectapp.js:10811 Xem does not have episode S12E17 for The Voice (247824),
using default format.app.js:10811 Xem does not have episode S12E16 for The
Voice (247824), using default format.app.js:10811 Xem does not have episode
S12E15 for The Voice (247824), using default format.app.js:10811 Xem does
not have episode S12E13 for The Voice (247824), using default
format.app.js:10818 Xem does not have series The Expanse (280619), using
default format.app.js:10818 Xem does not have series Deadliest Catch
(78957), using default format.app.js:10811 Xem does not have episode S00E16
for Bar Rescue (250071), using default format.app.js:10811 Xem does not
have episode S00E10 for Street Outlaws (270202), using default
format.zooqle.com/search?q=The%20Expanse%20S00E21%20&s=ns&sd=d&v=t
<http://zooqle.com/search?q=The%20Expanse%20S00E21%20&s=ns&sd=d&v=t> Failed
to load resource: the server responded with a status of 403
()www.skytorrents.in/search/all/ed/1/?q=The%20Expanse%20S00E21%20
<http://www.skytorrents.in/search/all/ed/1/?q=The%20Expanse%20S00E21%20>
Failed to load resource:
net::ERR_CONNECTION_TIMED_OUTzooqle.com/search?q=inside%20the%20expanse%3A%20season%202%20episode%2011%20&s=ns&sd=d&v=t
<http://zooqle.com/search?q=inside%20the%20expanse%3A%20season%202%20episode%2011%20&s=ns&sd=d&v=t>
Failed to load resource: the server responded with a status of 403
()zooqle.com/search?q=inside%20the%20expanse%3A%20&s=ns&sd=d&v=t
<http://zooqle.com/search?q=inside%20the%20expanse%3A%20&s=ns&sd=d&v=t>
Failed to load resource: the server responded with a status of 403
()zooqle.com/search?q=inside%20the%20expanse%20&s=ns&sd=d&v=t
<http://zooqle.com/search?q=inside%20the%20expanse%20&s=ns&sd=d&v=t> Failed
to load resource: the server responded with a status of 403
()www.skytorrents.in/search/all/ed/1/?q=inside%20the%20expanse%20
<http://www.skytorrents.in/search/all/ed/1/?q=inside%20the%20expanse%20>
Failed to load resource: net::ERR_CONNECTION_TIMED_OUTapp.js:379 GET
https://zooqle.com/search?q=The%20Voice%20US%20S12E13%20&s=ns&sd=d&v=t
<https://zooqle.com/search?q=The%20Voice%20US%20S12E13%20&s=ns&sd=d&v=t>
403 ()(anonymous) @ app.js:379n @ deps.js:2414(anonymous) @
deps.js:2412(anonymous) @ deps.js:2446$eval @ deps.js:2461$digest @
deps.js:2458$apply @ deps.js:2461(anonymous) @ deps.js:2595eg @
deps.js:2351d @ deps.js:2351app.js:379 GET
https://zooqle.com/search?q=The%20Voice%20US%20S12E1%20&s=ns&sd=d&v=t
<https://zooqle.com/search?q=The%20Voice%20US%20S12E1%20&s=ns&sd=d&v=t> 403
()(anonymous) @ app.js:379n @ deps.js:2414(anonymous) @
deps.js:2412(anonymous) @ deps.js:2446$eval @ deps.js:2461$digest @
deps.js:2458$apply @ deps.js:2461(anonymous) @ deps.js:2474f @
deps.js:2359(anonymous) @ deps.js:2362app.js:379 GET
https://zooqle.com/search?q=The%20Voice%20US%20S12E%20&s=ns&sd=d&v=t
<https://zooqle.com/search?q=The%20Voice%20US%20S12E%20&s=ns&sd=d&v=t> 403
()(anonymous) @ app.js:379n @ deps.js:2414(anonymous) @
deps.js:2412(anonymous) @ deps.js:2446$eval @ deps.js:2461$digest @
deps.js:2458$apply @ deps.js:2461(anonymous) @ deps.js:2474f @
deps.js:2359(anonymous) @ deps.js:2362app.js:379 GET
https://zooqle.com/search?q=The%20Voice%20US%20S12%20&s=ns&sd=d&v=t
<https://zooqle.com/search?q=The%20Voice%20US%20S12%20&s=ns&sd=d&v=t> 403
()(anonymous) @ app.js:379n @ deps.js:2414(anonymous) @
deps.js:2412(anonymous) @ deps.js:2446$eval @ deps.js:2461$digest @
deps.js:2458$apply @ deps.js:2461(anonymous) @ deps.js:2474f @
deps.js:2359(anonymous) @ deps.js:2362app.js:379 GET
https://zooqle.com/search?q=The%20Voice%20US%20S1%20&s=ns&sd=d&v=t
<https://zooqle.com/search?q=The%20Voice%20US%20S1%20&s=ns&sd=d&v=t> 403
()(anonymous) @ app.js:379n @ deps.js:2414(anonymous) @
deps.js:2412(anonymous) @ deps.js:2446$eval @ deps.js:2461$digest @
deps.js:2458$apply @ deps.js:2461(anonymous) @ deps.js:2474f @
deps.js:2359(anonymous) @ deps.js:2362deps.js:2433 Error -1:(anonymous) @
deps.js:2433(anonymous) @ deps.js:2406(anonymous) @ deps.js:2447$eval @
deps.js:2461$digest @ deps.js:2458(anonymous) @ deps.js:2461f @
deps.js:2359(anonymous) @ deps.js:2362app.js:379 GET
https://zooqle.com/search?q=The%20Voice%20US%20S%20&s=ns&sd=d&v=t
<https://zooqle.com/search?q=The%20Voice%20US%20S%20&s=ns&sd=d&v=t> 403
()(anonymous) @ app.js:379n @ deps.js:2414(anonymous) @
deps.js:2412(anonymous) @ deps.js:2446$eval @ deps.js:2461$digest @
deps.js:2458$apply @ deps.js:2461(anonymous) @ deps.js:2474f @
deps.js:2359(anonymous) @ deps.js:2362app.js:379 GET
https://zooqle.com/search?q=The%20Voice%20US%20&s=ns&sd=d&v=t
<https://zooqle.com/search?q=The%20Voice%20US%20&s=ns&sd=d&v=t> 403
()(anonymous) @ app.js:379n @ deps.js:2414(anonymous) @
deps.js:2412(anonymous) @ deps.js:2446$eval @ deps.js:2461$digest @
deps.js:2458$apply @ deps.js:2461(anonymous) @ deps.js:2474f @
deps.js:2359(anonymous) @ deps.js:2362deps.js:2433 Error -1:(anonymous) @
deps.js:2433(anonymous) @ deps.js:2406(anonymous) @ deps.js:2447$eval @
deps.js:2461$digest @ deps.js:2458(anonymous) @ deps.js:2461f @
deps.js:2359(anonymous) @ deps.js:2362app.js:379 GET
https://www.skytorrents.in/search/all/ed/1/?q=The%20Voice%20US%20
<https://www.skytorrents.in/search/all/ed/1/?q=The%20Voice%20US%20>
net::ERR_CONNECTION_TIMED_OUT(anonymous) @ app.js:379n @
deps.js:2414(anonymous) @ deps.js:2412(anonymous) @ deps.js:2446$eval @
deps.js:2461$digest @ deps.js:2458$apply @ deps.js:2461(anonymous) @
deps.js:2474f @ deps.js:2359(anonymous) @ deps.js:2362app.js:379 GET
https://zooqle.com/search?q=The%20Voice%20US%20S12E18%20&s=ns&sd=d&v=t
<https://zooqle.com/search?q=The%20Voice%20US%20S12E18%20&s=ns&sd=d&v=t>
403 ()(anonymous) @ app.js:379n @ deps.js:2414(anonymous) @
deps.js:2412(anonymous) @ deps.js:2446$eval @ deps.js:2461$digest @
deps.js:2458$apply @ deps.js:2461(anonymous) @ deps.js:2595eg @
deps.js:2351d @ deps.js:2351app.js:379 GET
https://www.skytorrents.in/search/all/ed/1/?q=The%20Voice%20US%20S12E18%20
<https://www.skytorrents.in/search/all/ed/1/?q=The%20Voice%20US%20S12E18%20>
net::ERR_CONNECTION_TIMED_OUT(anonymous) @ app.js:379n @
deps.js:2414(anonymous) @ deps.js:2412(anonymous) @ deps.js:2446$eval @
deps.js:2461$digest @ deps.js:2458$apply @ deps.js:2461(anonymous) @
deps.js:2595eg @ deps.js:2351d @ deps.js:2351*

**Comment by garfield69** [comment]
# Deluge diagnostics
Time to dig deeper.     
- shutdown DuckieTV and Deluge
- enable the Deluge log by adding the following to your shortcut target path ` -L info -l  "%UserProfile%\Desktop\deluge.log"`      
![image](https://cloud.githubusercontent.com/assets/6933240/25318572/fa96fbda-28e5-11e7-8136-9568384b0401.png)
- and startup Deluge
- Startup DuckieTV and press **F12** to open the *Development Tools*.
- Click on the console tab, and then the settings, and enable timestamps (useful for later)
![image](https://cloud.githubusercontent.com/assets/6933240/25318595/9be3f6aa-28e6-11e7-94ff-20f6abe5e4f0.png)
![image](https://cloud.githubusercontent.com/assets/6933240/25318612/c68dd330-28e6-11e7-8de0-275215dab42a.png)
![image](https://cloud.githubusercontent.com/assets/6933240/25318631/ed487f2a-28e6-11e7-91e3-9f353d014445.png)
- now click on the **Network** tab and leave this window open for now. This panel monitors all traffic in and out of DuckieTV
![image](https://cloud.githubusercontent.com/assets/6933240/25318644/28d34994-28e7-11e7-9de9-425e7a51ccba.png)
- back on the main DuckieTV page, find and submit a magnet, then look on the **console** for the event and note the time
![image](https://cloud.githubusercontent.com/assets/6933240/25318651/79c12fd8-28e7-11e7-8ae9-eaa20c8d9c04.png)
- on the Network page, examine the **json** entries and look for the event that matches the time. Here is an example of a good magnet submission
![image](https://cloud.githubusercontent.com/assets/6933240/25318675/f764817e-28e7-11e7-9ba3-bade43bf55fa.png)
- the status 200 means Deluge accepted the magnet post. Scroll down the record to check it is the right one, look for *web.add_torrents* and check the magnet hash is the same as the one on the console log
![image](https://cloud.githubusercontent.com/assets/6933240/25318730/b66d693c-28e8-11e7-962a-2d3afdd123ac.png)
- open the deluge log with a plain text editor and look for a corresponding entry there      
![image](https://cloud.githubusercontent.com/assets/6933240/25318747/f269b63e-28e8-11e7-8ff5-1facd5775dde.png)

what do you get?      

- Also, on the DuckieTV main page, click on the Deluge icon to open the Deluge Client monitor
![image](https://cloud.githubusercontent.com/assets/6933240/25318767/2f771832-28e9-11e7-84ad-3f330c0330ea.png)
- back at the Network page look for a json record with *web.update_ui*
![image](https://cloud.githubusercontent.com/assets/6933240/25318776/75366742-28e9-11e7-9601-a8e97fef5977.png)
- check for status 200, and look at the response, you should see something like this
![image](https://cloud.githubusercontent.com/assets/6933240/25318792/a026152e-28e9-11e7-8351-98a5eb8c1e84.png)
- there may be one or more entries, one for each torrent on deluge.     

what do you get?

# DuckieTV and XEM
Every time DuckieTV looks up its copy of the xem tables, it posts na info to the log. But for every pair of *series/episode* we lookup, we only post *once*, to avoid spamming the log.
- after any torrent search you can look to see if xem was used for the search or not, and what was done
 - open the console, and filter by xem
![image](https://cloud.githubusercontent.com/assets/6933240/25318858/8308c6d4-28ea-11e7-939e-4898780f6708.png)
- messages and meanings:
  - "Xem does not have series %s (%s), using default format."    
    The series is not being tracked by xem, the original s/e is used
  - "Xem has episode %s for %s (%s), using mapped format.     
    xem is tracking this series, and duckietv's current table has a cross reference. the s/e from xem is used instead of the original.
  - "Xem does not have episode %s for %s (%s), using default format."     
    xem is tracking the series, but duckietv's current table does not have a cross reference for this episode. the original s/e is used.


# analysis of your searches and comparison with mine

>The Expanse S00E21 (inside the expanse: season 2 episode 11)     

- the expanse is not tracked by xem
- none of the search engines that duckietv is using can find an torrent that contains *expanse s00e21*
- there are some torrents out there for *Inside the expanse*, but none for s02e11. The highest torrent found so far is for s02e09

OK. Special episodes are a category of their own, usually designated under *season 0*. You almost never be able to find torrents with the s00exx out there. You will almost never be able to use the auto-download to find special episodes. Generally you will need to do a manual search to tweek the search parameters. 

>the.voice.us.s12e14      

- the voice is tracked by xem, but they currently only have cross-references up to season 11.
- search for s12e14 yields just one torrent, which seven of duckietv's search engines are finding
![image](https://cloud.githubusercontent.com/assets/6933240/25319167/c70d3ae6-28ee-11e7-9737-09f43324564b.png)

>American Pickers specials     

already covered specials.

Hope this has been some use to you.

**Comment by cocokola** [comment]
Thanks for the reply, apologies for the late response.  I stalled after
realizing I had to reconfigure my firewall settings to allow the debug
version to work, then I fell quite ill.

When I recovered, I was sidetracked by trying to set up Transmission in its
own Linux container and I also setup sonarr and radarr.   The
Transmission had its own problems with throughput, and sonarr and
radarr although having more choices for sites to use, it did not support
most of the best trackers you have and very few seeders were
available making that solution nearly useless.  I was able to get the webui
of deluge to not crash with uninstalling/reinstalling; however, I still
need it to be running for Duckie to work.

I am curious, have you tested your setup with deluge with the default
password changed?  I am betting that is the problem.

I will setup debugging this weekend via your guide. I pulled down the new
nightly and will see if that fixed it, but  Stay tuned!
